src/arch/riscv64/CodeGen.zig

```diff
@@ -7785,48 +7785,71 @@ fn airAtomicLoad(func: *Func, inst: Air.Inst.Index) !void {
     const pt = func.pt;
     const zcu = pt.zcu;
     const atomic_load = func.air.instructions.items(.data)[@intFromEnum(inst)].atomic_load;
-    const order: std.builtin.AtomicOrder = atomic_load.order;
 
-    const ptr_ty = func.typeOf(atomic_load.ptr);
-    const elem_ty = ptr_ty.childType(zcu);
-    const ptr_mcv = try func.resolveInst(atomic_load.ptr);
+    const result: MCValue = result: {
+        const order: std.builtin.AtomicOrder = atomic_load.order;
 
-    const bit_size = elem_ty.bitSize(zcu);
-    if (bit_size > 64) return func.fail("TODO: airAtomicStore > 64 bits", .{});
+        // This instruction is must-lower. If the result is unused, we still need a fence.
+        if (func.liveness.isUnused(inst)) {
+            switch (order) {
+                .unordered, .monotonic => {},
+                .acquire, .seq_cst => {
```

Comment:
The prepending fence isn't needed for `acquire`.

Reply:
How did you determine this?

Reply:
It doesn't really make sense to have + https://zig.godbolt.org/z/bj36cj456

Reply:
Is there a particular reason we would want to do the load itself if the result is unused?

Reply:
I would say that I have two reasons for not eliding the load.
1. The load still needs to go through the TLB and can trigger faults, which is useful for debugging.
2. Although the RVWMO memory model does not specifically say there are side effects to monotonic loads, it could theoretically be possible for the load to have some side effects. Better be safe than sorry I think.